miio/chuangmi_ir.py

```diff
@@ -185,6 +204,79 @@ def get_indicator_led(self):
         return self.send("get_indicatorLamp")
 
 
+class ChuangmiRemote(ChuangmiIr):
+    """Class representing new type of Chuangmi IR Remote Controller identified by model
+    "chuangmi-remote-h102a03_".
+
+    The new controller uses different format for learned IR commands, which actually is
+    the old format but with additional layer of compression.
+    """
+
+    @classmethod
+    def pronto_to_raw(cls, pronto: str, repeats: int = 1) -> Tuple[str, int]:
+        """Takes a Pronto Hex encoded IR command and number of repeats and returns a
+        tuple containing a string encoded IR signal accepted by controller and
+        frequency. Supports only raw Pronto format, starting with 0000.
+
+        :raises ChuangmiIrException if heatshrink2 package is not installed.
+
+        :param str pronto: Pronto Hex string.
+        :param int repeats: Number of extra signal repeats.
+        """
+
+        if heatshrink2 is None:
+            raise ChuangmiIrException("heatshrink2 library is missing")
+        raw, frequency = super().pronto_to_raw(pronto, repeats)
```

Comment:
The docstring above mentions that only prontos starting with `0000` are supported. I think an exception should also be raised if the input is in invalid format.

Reply:
Also, the another class expects that repeats >= 0, that should also be the case here?